### ISSUE TITLE
Consider to add a specific USER to the Dockerfile

### ISSUE THREAD

**Issue by saw303** [micronaut-projects/micronaut-gradle-plugin#161]
The Micronaut Gradle plugin sets the `WORKDIR` to `/home/app` by default but runs the application in the context of the `root` user.

According to the [best practices](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#user) of Docker manual it would be great if the plugin would start the application in the user context of user `app` instead of `root`.

```Dockerfile
//.... base image
WORKDIR /home/app
COPY build/layers/libs /home/app/libs
COPY build/layers/resources /home/app/resources
COPY build/layers/application.jar /home/app/application.jar
RUN groupadd -r app && useradd --no-log-init -r -g app app && chown -R app:app /home/app
USER app
ENTRYPOINT ["java", "-jar", "/home/app/application.jar"]
```

**Comment by martinhynar** [comment]
Using Micronaut 2.5.6 and building docker image now fails with 

```
Step 6/8 : RUN addgroup app && adduser -G app app -D && chown -R app:app /home/app
 ---> Running in e4e5d7ef0028
/bin/sh: addgroup: command not found
```

I am using OracleLinux based base image and the command is `groupadd` (as suggested in the initial comment) not `addgroup`.

**Comment by ilopmar** [comment]
I used `addgroup` because we don't use Oracle Linux by default and `groupadd` doesn't work with openjdk-alpine

**Comment by martinhynar** [comment]
Same problem with Fedora and I assume that other distros will suffer with that too. IMO, it needs to be either configurable or somehow flexible (execute groupadd if the addgroup fails)

**Comment by ilopmar** [comment]
PRs welcome :)

**Comment by ilopmar** [comment]
Fixed by #221
